xblocks_contrib/html/html.py

```diff
@@ -1,81 +1,134 @@
     is_extracted = True
 
-    def resource_string(self, path):
-        """Handy helper for getting resources from our kit."""
-        return files(__package__).joinpath(path).read_text(encoding="utf-8")
-
-    # TO-DO: change this view to display your data your own way.
-    def student_view(self, context=None):
-        """
-        Create primary view of the HtmlBlock, shown to students when viewing courses.
-        """
-        if context:
-            pass  # TO-DO: do something based on the context.
-
-        frag = Fragment()
-        frag.add_content(
-            resource_loader.render_django_template(
-                "templates/html.html",
-                {
-                    "count": self.count,
-                },
-                i18n_service=self.runtime.service(self, "i18n"),
+    def get_html(self):
+        """Returns html required for rendering the block."""
+        if self.data:
+            data = self.data
+            user_id = (
+                self.runtime.service(self, "user")
+                .get_current_user()
+                .opt_attrs.get(ATTR_KEY_DEPRECATED_ANONYMOUS_USER_ID)
             )
+            if user_id:
+                data = data.replace("%%USER_ID%%", user_id)
+            data = data.replace("%%COURSE_ID%%", str(self.scope_ids.usage_id.context_key))
+            return data
+        return self.data
+
+    def index_dictionary(self):
+        xblock_body = super().index_dictionary()
+        # Removing script and style
+        html_content = re.sub(
+            re.compile(
+                r"""
+                    <script>.*?</script> |
+                    <style>.*?</style>
+                """,
+                re.DOTALL | re.VERBOSE,
+            ),
+            "",
+            self.data,
         )
-
-        frag.add_css(self.resource_string("static/css/html.css"))
-        frag.add_javascript(self.resource_string("static/js/src/html.js"))
+        html_content = escape_html_characters(html_content)
+        html_body = {
+            "html_content": html_content,
+            "display_name": self.display_name,
+        }
+        if "content" in xblock_body:
+            xblock_body["content"].update(html_body)
+        else:
+            xblock_body["content"] = html_body
+        xblock_body["content_type"] = "Text"
+        return xblock_body
+
+    def student_view_data(self, context=None):  # pylint: disable=unused-argument
+        """
+        Return a JSON representation of the student_view of this XBlock.
+        """
+        if getattr(settings, "FEATURES", {}).get(self.ENABLE_HTML_XBLOCK_STUDENT_VIEW_DATA, False):
+            return {"enabled": True, "html": self.get_html()}
+        else:
+            return {
+                "enabled": False,
+                "message": f'To enable, set FEATURES["{self.ENABLE_HTML_XBLOCK_STUDENT_VIEW_DATA}"]',
+            }
+
+    @XBlock.supports("multi_device")
+    def student_view(self, _context):
```

Comment:
@kdmccormick 
We have removed following method within this XBlock class.
Is it alright? What is it used for?
https://github.com/openedx/edx-platform/blob/master/xmodule/html_block.py#L100-L105

Reply:
Courses can be configured to have their content publicly available to logged-out users: https://docs.openedx.org/en/latest/site_ops/install_configure_run_guide/configuration/enable_public_course.html

public_view is used to show that anonymous view. In the case of HTML block, public_view is identical to the logged-in student_view: https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/courseware_api/views.py#L627-L628. 

Whenever a built-in block implements public_view, we should implement it the same in the extracted block.